spec/services/mediator_spec.rb

```diff
@@ -0,0 +1,55 @@
+RSpec.describe Mediator do
+  subject { described_class }
+
+  let(:collection) { create(:collection) }
+  let(:user) { create(:user) }
+  let(:file) { nil }
+  let(:filename) { nil }
+  let(:link) { nil }
+  let(:chunking_method) { :basic }
+  let(:chunking_profile) { create(:chunking_profile, method: chunking_method, size: 800) }
+  let(:doc) { Document.new(state: :created, link:, file:, filename:, chunking_profile:, collection:, user:) }
+
+  describe "#ingest document" do
+    after do
+      FetchWebDocumentJob.clear
+      ChunkDocumentJob.clear
```

Comment:
Makes sense. I've replace the `after` block with a single call to `Sidekiq::Worker.clear_all` at the end of each example, making a further assumption that no example will leave jobs behind for future use.

This doesn't address your earlier question about using job queue counting instead of more specific behaviour. I'm doing it this way because I don't want the `Mediator` tests to know what the jobs do per se, but only care that certain sequencing happens. For now Mediator is a very simple if/else. Soon after this PR however I plan to introduce a flow based on media types, and at that point the tests might not need to check for jobs.

Reply:
I think if you tested that Mediator is making the right call to these job classes, you wouldn't need to check what's in the sidekiq queue at all.

E.g.:

```
before do
  allow(FetchWebDocumentJob).to receive("perform_async")`.
end

it "something something" do
  subject.ingest
  expect(FetchWebDocumentJob).to have_received("perform_async").with(doc.id)`.
end
```

1. This checks that your class enqueues the right job (as your spec does now)
2. The job doesn't actually run (as with your spec now)
3. This checks that it passed the right arg (which your spec does not do now)
4. We don't need to remember to call anything in an `after` block or risk introducing flakiness (not true with your spec right now)

There are examples of this in `spec/lib/sidekiq_cron_schedule_spec.rb`.

I won't mention this again - if you really want to merge this way, that's cool with me. At this point I'll leave it up to you.

Reply:
I agree, that is much better than checking the jobs list. Thanks for the example, let me try.

Reply:
@oxaronick, take a look now. I was able to get rid of the job counting by using strategy above. 

I had to keep one use of `#drain` to test the flow through the fetch job to the chunk job. Does that make sense? From what I can tell, using Sidekiq in test queues the job but until I call `#drain` the job won't run.

Reply:
That makes sense - you don't actually want the job to run, do you?

Sidekiq should be in test mode just based on the `$RAILS_ENV`, so it should happily queue your jobs up and then die with a full queue at the end of the test run.

Reply:
I only want the jobs to run _when_ I need to test the sequencing. For e.g. `FetchWebDocumentJob` needs to run to then call `Mediator#ingest` to continue to chunking.

Reply:
Per our chat, see latest commit where we have a simpler `Mediator` test where it has no insight into the implementation of `FetchWebDocumentJob`, but rather tests it's inputs.